### PR TITLE
IGAPP-755: Show NotFoundError for invalid Deep Links

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: api-client/src/errors/NotFoundError.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { LocalNewsType, TuNewsType } from '../routes'
 
-type NotFoundType = 'category' | 'event' | 'poi' | 'offer' | 'disclaimer' | TuNewsType | LocalNewsType | 'route'
+type NotFoundType = 'city' | 'category' | 'event' | 'poi' | 'offer' | 'disclaimer' | TuNewsType | LocalNewsType | 'route'
```
Comment (SteffiStuffel) [inline api-client/src/errors/NotFoundError.ts:3]: Why don't we have 'language' here also?

**Review by SteffiStuffel — COMMENTED** [review]
Tested this for pois and got stuck in the loading screen, when trying to open a invalid poi, see attached video

https://user-images.githubusercontent.com/18463686/144980391-2be990aa-fa25-4fc9-bd7e-537d1a2b73d7.mp4

i.

File: api-client/src/errors/NotFoundError.ts (hunk 1 above)
Comment (Taggotty) [reply api-client/src/errors/NotFoundError.ts:3]: I guess because we treat `LanguageNotAvailable` differently.

**Comment by klinzo** [comment]
Tested on Android, works fine for me (also POIs). I get the same error all the time (requested page could not be found), not like in web configured for the different routes. But I think that is fine, no need to make it complicated. Nice work!

**Review by klinzo — APPROVED** [review]

**Comment by klinzo** [comment]
If you want you could add a release note like `Opening invalid links is not handled correctly` or similar

**Comment by Taggotty** [comment]
> Tested on Android, works fine for me (also POIs). I get the same error all the time (requested page could not be found), not like in web configured for the different routes. But I think that is fine, no need to make it complicated. Nice work!

This is because of the Failure component, as it currently only distinguishes the error on base of the Error Code. I guess it's a design and translation decision what should be done.

**Review by SteffiStuffel — APPROVED** [review]
Tested again. Works fine also for invalid pois.

**Pull request merged**